### ISSUE TITLE
Read a Vault token from $HOME/.vault-token

### ISSUE THREAD

**Issue by dmaze** [TerryHowe/ansible-modules-hashivault#12]
If you run `vault auth`, it will write its token (on Linux and MacOS at least) to `$HOME/.vault-token`.  It would be helpful if this module understood that file.

As a workaround, before running Ansible, I currently need to

```sh
export VAULT_TOKEN=$(cat ~/.vault-token)
```

I think at https://github.com/TerryHowe/ansible-modules-hashivault/blob/master/ansible/module_utils/hashivault.py#L14 a more involved default token check would work

```python
def get_default_token():
  """Get a default Vault token from an environment variable or a file."""
  if 'VAULT_TOKEN' in os.environ:
    return os.environ['VAULT_TOKEN']
  token_file = os.path.expanduser('~/.vault-token')
  if os.path.exists(token_file):
    with open(token_file, 'r') as f:
      return f.read()
  return ''

token = dict(required=False, default=get_default_token(), type='str', no_log=True)
```

I'm not 100% clear on the environment where this can run (_e.g._, will Ansible run this code on the target system?) or where the token lives on Windows hosts.

**Comment by TerryHowe** [comment]
Looks good.  Since you've done all the work already, do you want to throw together a PR?

**Comment by TerryHowe** [comment]
Released 2.11

**Comment by dmaze** [comment]
Sorry for going silent, I've just gotten to use this now and it's pretty helpful.  Thanks for getting it in!
